ctapipe/io/simteleventsource.py

```diff
@@ -471,13 +466,22 @@ def _fill_trigger_info(data, array_event):
             data.trigger.event_type = EventType.UNKNOWN
 
         data.trigger.tels_with_trigger = trigger["triggered_telescopes"]
+        if self.allowed_tels:
+            data.trigger.tels_with_trigger = np.intersect1d(
+                data.trigger.tels_with_trigger,
+                self.subarray.tel_ids,
+                assume_unique=True,
+            )
         central_time = parse_simtel_time(trigger["gps_time"])
         data.trigger.time = central_time
 
         for tel_id, time in zip(
             trigger["triggered_telescopes"], trigger["trigger_times"]
         ):
-            # telesocpe time is relative to central trigger in ns
+            if self.allowed_tels:
```

Comment:
combine to one if:

```
if self.allowed_tels and tel_id not in self.allowed_tels:
```

python won't evaluate the second half if the first is already not true

Reply:
I need to remember that one finally